yt_dlp/extractor/youtube.py

```diff
@@ -2552,6 +2581,7 @@ def append_client(client_name):
             elif self._is_agegated(pr):
                 append_client(f'{client}_agegate')
 
+        yield player_url
```

Comment:
yielding a totally unrelated value is confusing. I believe it is better to simply forego the laziness here. (It was made lazy only for code simplicity and `list` is being called on the output immediately)